core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -165,21 +169,62 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
      ****************************************/
 
     /**
-     * @notice Transfers ownership of the caller's current position to `newSponsorAddress`.
+     * @notice Requests to transfer ownership of the caller's current position to `newSponsorAddress`. Once the request liveness is passed,
+     * the sponsor can execute the transfer.
+     * @dev The liveness length is the same as the withdrawal liveness.
+     */
+    function requestTransfer() public onlyPreExpiration() {
+        PositionData storage positionData = _getPositionData(msg.sender);
+        require(positionData.transferRequestPassTimestamp == 0);
+
+        // Make sure the proposed expiration of this request is not post-expiry.
+        uint256 requestPassTime = getCurrentTime().add(withdrawalLiveness);
+        require(requestPassTime <= expirationTimestamp);
+
+        // Update the position object for the user.
+        positionData.transferRequestPassTimestamp = requestPassTime;
+
+        emit RequestTransfer(msg.sender);
+    }
+
+    /**
+     * @notice After a passed transfer request (i.e., by a call to `requestTransfer` and waiting
+     * `withdrawalLiveness`), transfers ownership of the caller's current position to `newSponsorAddress`.
      * @dev Transferring positions can only occur if the recipient does not already have a position.
      * @param newSponsorAddress is the address to which the position will be transferred.
      */
-    function transfer(address newSponsorAddress) public onlyPreExpiration() {
+    function transferPassedRequest(address newSponsorAddress) public onlyPreExpiration() {
         require(_getCollateral(positions[newSponsorAddress].rawCollateral).isEqual(FixedPoint.fromUnscaledUint(0)));
         PositionData storage positionData = _getPositionData(msg.sender);
         require(positionData.requestPassTimestamp == 0);
```

Comment:
Note that `transferPassedRequest` is blocked whenever there is a withdrawal request pending, but neither `requestTransfer` nor `cancelTransfer` are blocked.